cmd/object-handlers.go

```diff
@@ -689,6 +689,167 @@ func (api objectAPIHandlers) getObjectHandler(ctx context.Context, objectAPI Obj
 	})
 }
 
+// GetObjectAttributes ...
+func (api objectAPIHandlers) getObjectAttributesHandler(ctx context.Context, objectAPI ObjectLayer, bucket, object string, w http.ResponseWriter, r *http.Request) {
+	opts, valid := getAndValidateAttributesOpts(ctx, w, r, bucket, object)
+	if !valid {
+		return
```

Comment:
missing writeErrorResponse() ?

Reply:
The write to w (http.ResponseWriter) is happening within the function. Amazon added some custom fields to the error response and returning that would mean returning 4x variables or 1x variable and 1x new struct customized for this return. That's why writing to w inside the function felt cleaner.